File: James/230202_study/프로_136798_기사단원의무기.java
import java.util.*;

class Solution {
    public int solution(int number, int limit, int power) {
        int answer = 0;
        int[] arr = new int[number];
        
        for(int i = 1; i <= number; i++) {
            int cnt = 0;
            for(int j = 1; j * j <= i; j++) {
                if(j * j == i)
                    cnt++;
                
                else if(i % j == 0)
                    cnt += 2;
            }
            
            arr[i - 1] = cnt;
        }
        
        // System.out.println(Arrays.toString(arr));
        
        for(int i = 0; i < arr.length; i++) {
            if(arr[i] > limit)
                arr[i] = power;
        }
        
        // System.out.println(Arrays.toString(arr));
        
        for(int i : arr)
            answer += i;
        
        return answer;
    }
}

File: sammy/230208_study/프로_136798_기사단원의 무기.py
def yaksu(n):
    cnt=0
    # n의 제곱근까지의 약수 구하기
    for i in range(1,int(n**0.5)+1):
        # 나눠 떨어지면 cnt+1
        if n%i == 0:
            cnt+=1
            # 제곱했을 때 n이 되면 짝이 없는 것, n이 아니면 짝이 있으므로 cnt+1
            if i**2 != n:
                cnt+=1
    return cnt
            
def solution(number, limit, power):
    answer = 0
    for i in range(1,number+1):
        yaksu_cnt=yaksu(i)
        # 약수 개수가 limit를 넘으면 power을 더해주고, 그 외에는 약수 개수 더해주기 
        if yaksu_cnt>limit:
            answer+=power
        else:
            answer+=yaksu_cnt
    return answer

# print(solution(5,3,2))
# print(solution(10,3,2))